src-java/flowhs-topology/flowhs-storm-topology/src/main/java/org/openkilda/wfm/topology/flowhs/fsm/yflow/reroute/actions/HandleNotReroutedSubFlowAction.java

```diff
@@ -76,14 +86,31 @@ protected Optional<Message> performWithResponse(State from, State to, Event even
         }
 
         if (isFirstError) {
-            Message message = stateMachine.buildErrorMessage(context.getErrorType(), getGenericErrorMessage(),
-                    context.getError());
+            Message message;
+            if (ErrorType.ALREADY_ON_BEST_PATH.equals(context.getErrorType())) {
+                message = buildRerouteResponseMessage(stateMachine);
+            } else {
+                message = stateMachine.buildErrorMessage(context.getErrorType(), getGenericErrorMessage(),
+                        context.getError());
+            }
```

Comment:
Is this tested?

Reply:
thanks
cover with the unit tests all the corner cases